### PR TITLE
Add redux and redux saga to app

### PR DESCRIPTION
**Pull request by wley3337**
Adding in Redux and Redux-Saga plumbing.  Might have messed up some of the plumbing, but can't spin up `main` or this branch locally 🤷

### CODE REVIEW

File: src/redux/login/login.types.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export const LOGIN = 'LOGIN';
```
Comment (wley3337) [inline src/redux/login/login.types.ts:1]: Not sure if we want types in their own file or put them in with the actions?

File: src/redux/rootReducer.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { combineReducers } from 'redux';
+import { user } from './user/user.reducer';
+
+const rootReducer = combineReducers({ user });
+
+export default rootReducer;
+export type AppTate = ReturnType<typeof rootReducer>;
```
Comment (geekingreen) [inline src/redux/rootReducer.ts:7]: AppState?

**Review by geekingreen — COMMENTED** [review]
Looks good, I tested it out real quick and I could query the store and perform actions, just one comment on the name of an interface

**Comment by wley3337** [comment]
Yeah, missed that lol

**Pull request merged**